.github/workflows/hive.yml

```diff
@@ -0,0 +1,98 @@
+on:
+  push:
+    branches:
+      - main
+      - dan/hive-ci
+  pull_request:
+
+env:
+  RUSTFLAGS: -D warnings
+  CARGO_TERM_COLOR: always
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
+name: hive
+jobs:
+  build-reth:
+    name: build reth container
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v2
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Build and export reth image
+        uses: docker/build-push-action@v4
+        with:
+          context: .
+          tags: paradigmxyz/reth:main
+          outputs: type=docker,dest=/tmp/reth_image.tar
+
+      # TODO: turn this into an actual image publish
+      - name: Upload reth image
+        uses: actions/upload-artifact@v3
+        with:
+          name: reth_image
+          path: /tmp/reth_image.tar
+
+
+  hive-matrix:
```

Comment:
We should add a cache here for the `go build` steps

Reply:
Good idea, that should speed things up a bit